src/GafferSceneUI/LightVisualiser.cpp

```diff
@@ -0,0 +1,188 @@
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "boost/container/flat_map.hpp"
+#include "boost/algorithm/string/predicate.hpp"
+
+#include "IECoreGL/CurvesPrimitive.h"
+#include "IECoreGL/Group.h"
+
+#include "GafferSceneUI/LightVisualiser.h"
+#include "GafferSceneUI/AttributeVisualiser.h"
+#include "GafferSceneUI/StandardLightVisualiser.h"
+
+using namespace std;
+using namespace Imath;
+using namespace GafferSceneUI;
+
+//////////////////////////////////////////////////////////////////////////
+// Internal implementation details
+//////////////////////////////////////////////////////////////////////////
+
+namespace
+{
+
+typedef boost::container::flat_map<IECore::InternedString, ConstLightVisualiserPtr> LightVisualisers;
+LightVisualisers &lightVisualisers()
+{
+	static LightVisualisers l;
+	return l;
+}
+
+const LightVisualiser *standardLightVisualiser()
+{
+	static ConstLightVisualiserPtr l = new StandardLightVisualiser;
+	return l.get();
+}
+
+/// Class for visualisation of lights. All lights in Gaffer are represented
+/// as IECore::Light objects, but we need to visualise them differently
+/// depending on their shader name (accessed using `IECore::Light::getName()`). A
+/// factory mechanism is provided to map from this type to a specialised
+/// LightVisualiser.
+class AttributeVisualiserForLights : public AttributeVisualiser
+{
+
+    public :
+
+        IE_CORE_DECLAREMEMBERPTR( AttributeVisualiserForLights )
+
+        typedef IECore::Light ObjectType;
```

Comment:
I think this typedef dates back to when lights were ObjectVisualisers and can be removed?